handlers/ruleset.go

```diff
@@ -23,90 +23,252 @@ package handlers
 import (
 	"errors"
 	"fmt"
+	"time"
 
 	"github.com/m3db/m3cluster/kv"
 	"github.com/m3db/m3metrics/generated/proto/schema"
+	"github.com/m3db/m3metrics/policy"
+	"github.com/m3db/m3metrics/rules"
+	"github.com/pborman/uuid"
 )
 
 var (
 	errMultipleMatches = errors.New("more than one match found")
+	errRuleSetExists   = errors.New("ruleset already exists")
+	errNoPolicies      = errors.New("no policies provided")
+	errNoTargets       = errors.New("no targets provided")
+	errNoFilters       = errors.New("no filters provided")
 )
 
-// Rule returns the rule with a given name, or an error if there are mutliple matches
-func Rule(ruleSet *schema.RuleSet, ruleName string) (*schema.MappingRule, *schema.RollupRule, error) {
-	var (
-		mappingRule *schema.MappingRule
-		rollupRule  *schema.RollupRule
-	)
-	for _, mr := range ruleSet.MappingRules {
-		if len(mr.Snapshots) == 0 {
-			continue
-		}
+//RollupTarget is a representation of rules.RollupTarget where the policies are
+//still in strings format.
+type RollupTarget struct {
+	Name     string   `json:"name"`
+	Tags     []string `json:"tags"`
+	Policies []string `json:"policies"`
+}
 
-		latestSnapshot := mr.Snapshots[len(mr.Snapshots)-1]
-		name := latestSnapshot.Name
-		if name != ruleName || latestSnapshot.Tombstoned {
-			continue
-		}
-		if mappingRule == nil {
-			mappingRule = mr
-		} else {
-			return nil, nil, errMultipleMatches
+// TODO(dgromov): Return aggregate errors instead of the first one.
```

Comment:
Yeah, it'd be nice to do at some point. Right now if you have multiple invalid policies, you'll only know about the first one.